libs/variant/include/variant/variant.hpp

```diff
@@ -576,4 +584,167 @@ void Variant::IterateObject(Function const &function) const
 }
 
 }  // namespace variant
+
+namespace serializers {
+
+template <typename D>
+struct ForwardSerializer<fetch::variant::Variant, D>
+{
+public:
+  using Type       = fetch::variant::Variant;
+  using DriverType = D;
+
+  template <typename Serializer>
+  static void Serialize(Serializer &serializer, Type const &var)
+  {
+    auto typecode = static_cast<int>(var.type());
+    serializer << typecode;
+    switch (var.type())
+    {
+    case Type::Type::UNDEFINED:
+    {
+      return;
+    }
+
+    case Type::Type::NULL_VALUE:
+    {
+      return;
+    }
+    case Type::Type::INTEGER:
+    {
+      serializer << var.As<int>();
+      return;
+    }
+    case Type::Type::FLOATING_POINT:
+    {
+      serializer << var.As<double>();
+      return;
+    }
+    case Type::Type::FIXED_POINT:
+    {
+      serializer << var.As<int>();
+      return;
+    }
+    case Type::Type::BOOLEAN:
+    {
+      serializer << var.As<bool>();
+      return;
+    }
+    case Type::Type::STRING:
+    {
+      serializer << var.As<std::string>();
+      return;
+    }
+    case Type::Type::ARRAY:
+    {
+      auto sz = static_cast<uint32_t>(var.size());
+      serializer << sz;
+      for(std::size_t i=0;i<var.size();i++)
+      {
+        Serialize(serializer, var[i]);
+      }
+      return;
+    }
+    case Type::Type::OBJECT:
+    {
+      auto sz = static_cast<uint32_t>(var.size());
+      serializer << sz;
+      var.IterateObject([&](auto const &k, auto const &v){
+          serializer << k;
+          Serialize(serializer, v);
+          return true;
+        });
+      return;
+    }
+    };
+    throw std::runtime_error{"Variant has unknown type."};
+  }
+
+  template <typename Deserializer>
+  static void Deserialize(Deserializer &deserializer, Type &var)
+  {
+    int typecode;
+    deserializer >> typecode;
+    auto type = static_cast<Type::Type>(typecode);
+    switch (type)
+    {
+    case Type::Type::UNDEFINED:
+      {
+        var = fetch::variant::Variant::Undefined();
+        return;
+      }
+    case Type::Type::NULL_VALUE:
+      {
+        var = fetch::variant::Variant::Null();
+        return;
+      }
+    case Type::Type::INTEGER:
+      {
+        int64_t tmp;
+        deserializer >> tmp;
+        var = tmp;
+        return;
+      }
+    case Type::Type::FLOATING_POINT:
+      {
+        double tmp;
+        deserializer >> tmp;
+        var = tmp;
+        return;
+      }
+    case Type::Type::FIXED_POINT:
+      {
+        int64_t tmp;
```

Comment:
I would suggest to use the `fixed_point::fp64_t tmp;`, There will be no performance hit, since all what `FixedPoint<32,32>` class contains is just `uint64_t` as data member.

Reply:
done